ngraph_bridge/ngraph_builder.cc

```diff
@@ -976,7 +976,8 @@ static Status TranslateBatchMatMulV2Op(
           "for each dimension. Failed to match dimension ",
           i, " where lhs was ", ng_lhs_shape[i], " and rhs was ",
           ng_rhs_shape[i]);
-    } else if ((ng_lhs_shape[i] == 1) || (ng_rhs_shape[i] == 1)) {
+    } else if (((ng_lhs_shape[i] == 1) || (ng_rhs_shape[i] == 1)) &&
+               (ng_lhs_shape[i] != ng_lhs_shape[i])) {
       return errors::Unimplemented(
           "ng_lhs_shape and ng_rhs_shape must be the same for each dimension, "
           "for current implementation of BatchMatMulV2."
```

Comment:
Also, put  a space at the beginning of the next line: "Failed to match dimension ",

Reply:
Done.